deepchem/models/tensorgraph/models/graph_models.py

```diff
@@ -603,7 +604,45 @@ def default_generator(self,
           d[self.deg_adjs[i - 1]] = multiConvMol.get_deg_adjacency_lists()[i]
         yield d
 
-  def predict_proba_on_generator(self, generator, transformers=[]):
+  def predict_on_generator(self, generator, transformers=[], outputs=None):
```

Comment:
A lot of this looks identical to the inherited superclass method. Why do we need the additional method here?

Reply:
https://github.com/deepchem/deepchem/blob/master/deepchem/models/tensorgraph/tensor_graph.py#L288 for reference

Reply:
Line 622 & 633 are the only change: current TensorgraphGraphConv outputs fixed size output(of `batch_size`) regardless of input batch size, which is due to the `tf.dynamic_partition` operation in GraphGather layer. This `n_samples` records the input size and cutoff the extra parts in output correspondingly.

Reply:
Ah I see. Perhaps this fix should be made directly to `TensorGraph` then? Looks pretty general

Reply:
I would like to, but currently the batch size is inferred from an input feature placeholder for GraphConv(not existing in other methods). I have tried to use label or weight placeholder, but they cause exceptions in GeneratorEvaluator part(it will remove all labels and weights input before model inference). I feel this is the most neutral way to do it since the problem only happens to GraphConv.

Reply:
Ah, ok I see. Would you mind adding some comments explaining this issue in the code? We should be good to merge in after that.

Reply:
Sweet, comments added, thanks for the review!